src/core/CoolBoard.cpp

```diff
@@ -377,11 +379,44 @@ void CoolBoard::readBoardData(JsonObject &reported) {
   }
 }
 
-void CoolBoard::sleep(unsigned long interval) {
-  if (interval > 0) {
-    INFO_VAR("Going to sleep for seconds:", interval);
-    ESP.deepSleep((uint64(interval) * 1000000ULL), WAKE_RF_DEFAULT);
+void CoolBoard::sleep() {
+  EEPROM.begin(5);
+  delay(5000);
+  uint8_t val[4];
+  if (!EEPROM.read(0x04) ||
+      digitalRead(BOOTSTRAP_PIN) == LOW) {
```

Comment:
Given that we can know the reset reason, I'd rather use [getResetInfoPtr](https://github.com/esp8266/Arduino/blob/63ab79e549bbf63b99cd53688ff1d253bb1a2671/cores/esp8266/Esp.cpp#L403-L405) to find out wether the reset was triggered by hitting the reset button

Reply:
See [`getResetReason`](https://github.com/esp8266/Arduino/blob/63ab79e549bbf63b99cd53688ff1d253bb1a2671/cores/esp8266/Esp.cpp#L372-L392) about usage of the `rst_info` data structure